sentinel/test/mocha/lib/message-utils.js

```diff
@@ -0,0 +1,331 @@
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(sms);
+        expect(messages[0].original_message).to.equal(undefined);
+        done();
+      });
+
+      it('does not truncate short unicode sms', done => {
+        const sms = generateMessage(MAX_UNICODE_LENGTH, true);
+        const config = { multipart_sms_limit: 10 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(sms);
+        expect(messages[0].original_message).to.equal(undefined);
+        done();
+      });
+
+      it('truncates long sms', done => {
+        const sms = generateMessage(1000);
+        const expected = sms.substr(0, 150) + '...';
+        const config = { multipart_sms_limit: 1 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(expected);
+        expect(messages[0].original_message).to.equal(sms);
+        done();
+      });
+
+      it('truncates long unicode sms', done => {
+        const sms = generateMessage(1000, true);
+        const expected = sms.substr(0, 64) + '...';
+        const config = { multipart_sms_limit: 1 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(expected);
+        expect(messages[0].original_message).to.equal(sms);
+        done();
+      });
+
+    });
+
+  });
+
+  describe('template', () => {
+
+    it('plain text', done => {
+      const actual = utils.template({}, null, {}, { message: 'hello' });
+      expect(actual).to.equal('hello');
+      done();
+    });
+
+    it('variables', done => {
+      const actual = utils.template({}, null, { name: 'george' }, { message: 'hello {{name}}' });
+      expect(actual).to.equal('hello george');
+      done();
+    });
+
+    describe('dates', () => {
+
+      it('string', done => {
+        const date = '2016-03-06T03:45:41.000Z';
+        const input = '{{#date}}{{reported_date}}{{/date}}';
+        const doc = { reported_date: date };
+        const config = { date_format: 'DD-MMM-YYYY' };
+        const actual = utils.template(config, null, doc, { message: input });
+        expect(actual).to.equal(moment(date).format(config.date_format));
```

Comment:
Seems a bit weird to use moment here to generate our expected output.  Can't we write a specific string here?

Reply:
I agree but I think there'd be timezone issues where the library would output a different string for the same UTC date depending on where you ran the test.